src/features/island/bumpkin/BumpkinBuilder.tsx

```diff
@@ -0,0 +1,258 @@
+  shirt: getRandomPart<LimitedShirt>("shirt"),
+  mouth: "Smile",
+  pants: "Farmer Pants",
+  shoes: "Black Shoes",
+});
+
+const findSelectedOptionIndex = (
+  selectedOption: string,
+  category: Category
+) => {
+  const { options } = BUMPKIN_PARTS[category];
+
+  return options.findIndex((option) => option === selectedOption);
+};
+
+interface Props {
+  onMint: () => void;
+}
+
+export const BumpkinBuilder: React.FC<Props> = ({ onMint }) => {
+  const [activeCategory, setActiveCategory] = useState<CategoryDetails>(
+    BUMPKIN_PARTS.hair
+  );
+  const [selectedOptionIndex, setSelectedOptionIndex] = useState(0);
+  const [bumpkinParts, setBumpkinParts] = useState<Bumpkin>(
+    makeInitialBumpkin()
+  );
+
+  useEffect(() => {
+    setSelectedOptionIndex(
+      findSelectedOptionIndex(
+        bumpkinParts[activeCategory.name],
+        activeCategory.name
+      )
+    );
+  }, []);
+
+  const handleSelect = (category: Category) => {
+    setActiveCategory(BUMPKIN_PARTS[category]);
+
+    setSelectedOptionIndex(
+      findSelectedOptionIndex(
+        bumpkinParts[BUMPKIN_PARTS[category].name],
+        BUMPKIN_PARTS[category].name
+      )
+    );
+  };
+
+  const handleArrowClick = (direction: "left" | "right") => {
+    const { options } = activeCategory;
+    let index = selectedOptionIndex;
+
+    switch (direction) {
+      case "left":
+        index--;
+        break;
+      case "right":
+        index++;
+        break;
+    }
+
+    if (index < 0) {
+      index = options.length - 1;
+    } else if (index === options.length) {
+      index = 0;
+    }
+
+    setSelectedOptionIndex(index);
+    setBumpkinParts({
+      ...bumpkinParts,
+      [activeCategory.name]: activeCategory.options[index],
+    });
+  };
+
+  const getImageToShow = (part: keyof Bumpkin) => {
+    // If rotating through a categories items then get the image from that
+    // categories options array
+    if (activeCategory.name === part) {
+      return ITEM_IMAGES[activeCategory.options[selectedOptionIndex]];
+    }
+    // else get the image from the selected bumpkin items
+    return ITEM_IMAGES[bumpkinParts[part]];
+  };
+
+  const handleMint = () => {
+    console.log(bumpkinParts);
+    onMint();
+  };
+
+  return (
+    <>
+      <div className="relative flex flex-col items-center p-2">
+        <h1 className="mb-2 text-">Bumpkin Builder</h1>
+        <div className="flex mb-2">
+          {getKeys(BUMPKIN_PARTS).map((category) => (
+            <Box
+              key={category}
+              image={BUMPKIN_PARTS[category].icon}
+              isSelected={activeCategory.name === category}
+              onClick={() => handleSelect(category)}
+            />
+          ))}
+        </div>
+        <div className="relative w-full">
+          <img
+            src={dropShadow}
+            alt="drop-shadow"
+            className="absolute bottom-0 z-0 opacity-30"
+          />
+          {getKeys(bumpkinParts).map((part, index) => (
```

Comment:
Similar to the `DynamicNPC` component, it could be useful to extract all of the image rendering into a separate component `DynamicNFT`.

This can be reused in places where we need to display the NFT version (i.e. Land Expansion Modal)